src/python/pants/backend/python/tasks2/pytest_run.py

```diff
@@ -592,7 +592,8 @@ def parse_error_handler(parse_error):
         raise TaskError('Error parsing xml file at {}: {}'
           .format(parse_error.xml_path, parse_error.cause))
 
-      all_tests_info = self.parse_test_info(junitxml_path, parse_error_handler, ['file', 'name'])
+      all_tests_info = self.parse_test_info(junitxml_path, parse_error_handler,
+                                            ['file', 'name', 'classname'])
```

Comment:
A [test exists](https://github.com/pantsbuild/pants/blob/master/tests/python/pants_test/task/test_testrunner_task_mixin.py#L521-L576) that exercises the case were we pass a field name to parse but it doesn't exist. Currently we are leaving it as an empty string if the testcase attribute doesn't exist. I can change this so that we are skipping the attribute if you'd like...